CMakeLists.txt

```diff
@@ -116,6 +116,7 @@ option(USE_STATIC_LIBSTDCXX "Should use static rather than shared libstdc++" OFF
 option(ENABLE_INET_PTON "Set to OFF to prevent usage of inet_pton when building against modern SDKs while still requiring compatibility with older Windows versions, such as Windows XP, Windows Server 2003 etc." ON)
 option(ENABLE_CODE_COVERAGE "Enable code coverage reporting" OFF)
 option(ENABLE_MONOTONIC_CLOCK "Enforced clock_gettime with monotonic clock on GC CV /temporary fix for #729/" OFF)
+option(ENABLE_STDCXX_CHRONO "Use C++11 std::chrono for timing instead of pthreads" OFF)
```

Comment:
In this form, this option would be at best intermediate. We simply need an option that allows to select from two alternatives:
* C++11 implementation for sensitive parts (threads and timers)
* C++03 emulated implementation

Might be that this option can get a string that would specify some details, that is, for example, threads, but not timers.

Reply:
Not sure about "thread but not timers". This would apply unnecessary restrictions to the implementation.
We have `ENABLE_CXX11` option, but it is used to enable apps. I doubt we can reuse it.

Reply:
So, if we can't turn on "threads, but not timers", that is, we can either use the C++11 implementation of the sync-related things, or the C++03 emulated version, then there should be simply an option to use C++11 sync. The `ENABLE_CXX11` option cannot be reused for that purpose, at best it can cooperate. Maybe `ENABLE_STDCXX_SYNC` ?

Reply:
`ENABLE_STDCXX_SYNC` sounds good!

There is also one thing, that if C++17 is enabled, then `scoped_lock` could be used instead of `lock_guard`. But we should probably just stay with `lock_guard`.

Reply:
That's something we can change later, if it does us any good.

Reply:
Added `ENABLE_STDCXX_SYNC` build option